### PR TITLE
I1742 & I1743 apply continuation header from syncv2 and download block sections

### PR DESCRIPTION
**Pull request by pragmaxim**
Fixes https://github.com/ergoplatform/ergo/issues/1742 #1743

### CODE REVIEW

**Comment by pragmaxim** [comment]
the biggest problem in the Synchronizer test suite is that this https://github.com/ergoplatform/ergo/blob/3fa0ed3d73bcc7d1e0130a347b182b8f5313f49e/src/test/scala/org/ergoplatform/network/ErgoNodeViewSynchronizerSpecification.scala#L100
history is used in Synchronizer but not in ErgoNodeViewHolder, which reads it from disk empty and we cannot test effects of Synchronizer on history done in ErgoNodeViewHolder

**Comment by kushti** [comment]
> @kushti the biggest problem in the Synchronizer test suite is that this
> 
> https://github.com/ergoplatform/ergo/blob/3fa0ed3d73bcc7d1e0130a347b182b8f5313f49e/src/test/scala/org/ergoplatform/network/ErgoNodeViewSynchronizerSpecification.scala#L100
> 
> history is used in Synchronizer but not in ErgoNodeViewHolder, which reads it from disk empty and we cannot test effects of Synchronizer on history done in ErgoNodeViewHolder

Yes, history is different in Synchronizer and ENVH, that's why I made `Synchronizer2Fixture` in 
```
property("NodeViewSynchronizer: receiving out-of-order header should request it again") {
    withFixture2 { ctx =>
```

File: src/test/scala/org/ergoplatform/network/ErgoNodeViewSynchronizerSpecification.scala [hunk 1]
```diff
@@ -236,37 +235,87 @@ class ErgoNodeViewSynchronizerSpecification extends HistoryTestHelpers with Matc
...
       }
 
       deliveryTracker.reset()
 
       // we generate fork of two headers, starting from the parent of the best header
       // so the depth of the rollback is 1, and the fork bypasses the best chain by 1 header
       val hhistory = ErgoHistory.readOrGenerate(settings, timeProvider)
-      val parent = hhistory.lastHeaders(2).head
+      val baseChain = genHeaderChain(_.size > 2, None, hhistory.difficultyCalculator, None, false)
+      baseChain.headers.foreach(sendHeader)
```
Comment (pragmaxim) [inline src/test/scala/org/ergoplatform/network/ErgoNodeViewSynchronizerSpecification.scala:306]: @kushti I made this change in out-of-order test as it fails on CI with empty `.ergo_test/history` folder ... it only works if some preceding tests generates some headers apriory.

File: src/main/scala/org/ergoplatform/nodeView/history/ErgoHistoryReader.scala [hunk 2]
```diff
@@ -266,6 +266,26 @@ trait ErgoHistoryReader
...
+    */
+  def continuationHeaderV2(syncInfo: ErgoSyncInfoV2): Option[Header] = {
+    if (syncInfo.lastHeaders.isEmpty) {
+      Option.empty
+    } else {
+      // let's find most recent header that is common to our history
+      commonPoint(syncInfo.lastHeaders) match {
+        case Some(commonHeader) =>
+          // now let's find continuation header whose parent is the common header
+          syncInfo.lastHeaders.find(_.parentId == commonHeader.id)
```
Comment (kushti) [inline src/main/scala/org/ergoplatform/nodeView/history/ErgoHistoryReader.scala:282]: In the real network (i.e. unlike tests) headers in syncv2 are not consecutive (they are located at offsets 0,16,128,512 from the chain tip), so this code will never succeed 

I think should be like

```
  if (syncInfo.lastHeaders.isEmpty) {
      Option.empty
    } else {
      val lastHeader = syncInfo.lastHeaders.head
      if (bestHeaderIdOpt.contains(lastHeader.parentId)) {
        Some(lastHeader)
      } else {
        Option.empty
      }
    }
```

**Review by kushti — CHANGES_REQUESTED** [review]

File: src/main/scala/org/ergoplatform/nodeView/history/ErgoHistoryReader.scala (hunk 2 above)
Comment (pragmaxim) [reply src/main/scala/org/ergoplatform/nodeView/history/ErgoHistoryReader.scala:282]: @kushti I was under the impression that the `syncV2` headers would rarely begin with our node's `bestHeaderIdOpt` so I was trying to find the `commonPoint` and apply header whose parent is the common point.

**Review by kushti — APPROVED** [review]

**Pull request merged**